### PR TITLE
fix: minor bug fixes and validations

### LINKED ISSUE #693: Interactive instance creation should check if the name exists during the input validation
Current behaviour:

```
❯ ? Name: test-instance
? Cloud Provider: aws
? Cloud Region: us-east-1
Error: Kafka cluster name is already used
```

Expected behaviour:
```
❯ ? Name: test-instance
Error: Kafka cluster name is already used
```

### PR DESCRIPTION
**Pull request by rkpattnaik780**
Closes #693, #694, #695

### Verification Steps

1. Do x
2. Do y

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### Checklist

- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: pkg/kafka/kafka_util.go [hunk 1]
```diff
@@ -82,3 +83,18 @@ func ValidateSearchInput(val interface{}) error {
 
 	return kafkaerr.InvalidSearchValueError(search)
 }
+
+// ValidateNameIsAvailable checks if a kafka instance with the given name already exists
+func ValidateNameIsAvailable(api kafkamgmtv1.DefaultApi) func(v interface{}) error {
+	return func(v interface{}) error {
+		name, _ := v.(string)
+
+		_, httpRes, _ := GetKafkaByName(context.Background(), api, name)
```
Comment (rkpattnaik780) [inline pkg/kafka/kafka_util.go:92]: Getting a lint error here with the message: `response body must be closed (bodyclose)`. Calling close() method does remove this message but causes other issues. Any suitable workaround for this is appreciated.

**Review by craicoverflow — CHANGES_REQUESTED** [review]
Missing verification steps made it a bit more difficult o verify quickly

Overall, the changes are great, but a couple of small changes are required.

File: pkg/kafka/kafka_util.go [hunk 2]
```diff
@@ -82,3 +83,18 @@ func ValidateSearchInput(val interface{}) error {
...
+
+		if httpRes != nil && httpRes.StatusCode == 200 {
+			return errors.New("kafka cluster name is already used")
```
[the remark this answers is not in the document]
Comment (rkpattnaik780) [reply pkg/kafka/kafka_util.go:95]: Non-interactive mode returns `Kafka cluster` (error message from server).
Screenshot from 2021-06-01 19-34-47
It should be changed either from the API first or should we define a set of error messages for each status code(as done for `kafka topic`)?

[the remark this answers is not in the document]
Comment (craicoverflow) [reply pkg/kafka/kafka_util.go]: We should do the latter. We probably do not need to do it for all of them yet, but for this one we should yeah.

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**